master/internal/api_task.go

```diff
@@ -30,3 +30,28 @@ func (a *apiServer) GetTask(
 			errors.Wrapf(err, "error fetching task %s from database", req.TaskId)
 	}
 }
+
+func (a *apiServer) GetGenericTaskConfig(
+	ctx context.Context, req *apiv1.GetGenericTaskConfigRequest,
+) (resp *apiv1.GetGenericTaskConfigResponse, err error) {
+	if err := a.canDoActionsOnTask(ctx, model.TaskID(req.TaskId),
+		expauth.AuthZProvider.Get().CanGetExperimentArtifacts); err != nil {
```

Comment:
I the logic actually executes CanGetNSC, `CanGetExperimentArtifacts` only gets executed for Trials. I removed the function input to avoid confusion here as it would never execute.

Reply:
okay, makes sense. `canDoActionsOnTask` has somewhat unconventional flow so I was confused